File: 075.Sort Colors.cpp
class Solution {
public:
    void sortColors(vector<int>& nums) {
        if (nums.size() < 2)
            return;
        
        int colorLeft = 0;
        int colorRight = nums.size() - 1;
        int i = 0;
        
        while (i <= colorRight) {
            if (nums[i] == 0) {
                swap(nums[colorLeft], nums[i]);
                colorLeft++;
                i++;
            } else if (nums[i] == 1) {
                i++;
            } else {
                swap(nums[colorRight], nums[i]);
                colorRight--;
            }
        }
    }
};

File: 075. Sort Colors.py
class Solution:
    def sortColors(self, nums):
        """
        :type nums: List[int]
        :rtype: void Do not return anything, modify nums in-place instead.
        """
        left = 0
        right = len(nums) - 1
        curt = 0
        
        while curt <= right:
            if nums[curt] == 0:
                nums[left], nums[curt] = nums[curt], nums[left]
                left += 1
                curt += 1
            elif nums[curt] == 2:
                nums[right], nums[curt] = nums[curt], nums[right]
                right -= 1
            else:
                curt += 1
